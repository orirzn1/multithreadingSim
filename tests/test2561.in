L13
S7
O5
N7

T0
I@0x00000000
STORE $4, $7, $7
LOAD $4, $6, $3
ADD $0, $7, $5
ADD $3, $2, $7
LOAD $0, $2, $3
ADD $0, $2, $6
SUB $3, $1, $7
ADDI $3, $4, 0x1
SUBI $3, $7, 0x1
SUBI $5, $7, 0x2
HALT $0

T1
I@0x00000000
SUBI $0, $4, 0x1
LOAD $3, $6, $0
LOAD $5, $0, 0x1
STORE $0, $6, 0x4
HALT $0

T2
I@0x00000000
LOAD $2, $7, 0x3
ADD $2, $1, $2
LOAD $2, $4, $0
SUB $7, $1, $1
LOAD $2, $0, 0x0
STORE $5, $3, $7
SUB $2, $6, $1
SUB $4, $1, $2
STORE $1, $3, 0x4
HALT $0

T3
I@0x00000000
SUB $6, $6, $2
SUBI $4, $3, 0x4
ADD $2, $7, $5
SUBI $7, $4, 0x1
ADD $6, $0, $0
SUBI $2, $5, 0x4
HALT $0

T4
I@0x00000000
SUB $0, $7, $4
SUBI $1, $5, 0x3
STORE $2, $5, $2
SUBI $3, $5, 0x3
SUBI $3, $5, 0x4
ADDI $3, $0, 0x3
ADD $0, $1, $0
STORE $4, $7, $3
HALT $0

T5
I@0x00000000
SUBI $5, $4, 0x2
SUB $5, $4, $5
ADD $5, $7, $6
STORE $3, $5, 0x0
LOAD $6, $4, $6
ADDI $7, $3, 0x3
SUBI $5, $4, 0x1
SUBI $7, $3, 0x2
LOAD $7, $1, 0x1
HALT $0

T6
I@0x00000000
ADD $2, $1, $2
ADD $1, $4, $5
LOAD $0, $6, $3
ADD $4, $5, $4
LOAD $2, $6, $7
STORE $0, $0, $2
STORE $5, $2, 0x4
HALT $0

D@0x0
0x9
0x4
0x7
0x6
0x7
0x8
0x7
0x2
0x6
0x4
