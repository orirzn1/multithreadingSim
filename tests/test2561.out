
---- Blocked MT Simulation ----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0x0	R5 = 0xFFFFFFFE	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFB
Register file thread id 4:
	R0 = 0xFFFFFFFD	R1 = 0xFFFFFFFD	R2 = 0x0	R3 = 0x3	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFF	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Blocked MT CPI for this program 3.183333

-----Finegrained MT Simulation -----

Register file thread id 0:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0xFFFFFFFF	R4 = 0x0	R5 = 0xFFFFFFFE	R6 = 0x0	R7 = 0x0
Register file thread id 1:
	R0 = 0xFFFFFFFF	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 2:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 3:
	R0 = 0x0	R1 = 0x0	R2 = 0xFFFFFFFC	R3 = 0x0	R4 = 0xFFFFFFFC	R5 = 0x0	R6 = 0x0	R7 = 0xFFFFFFFB
Register file thread id 4:
	R0 = 0xFFFFFFFD	R1 = 0xFFFFFFFD	R2 = 0x0	R3 = 0x3	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Register file thread id 5:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0xFFFFFFFF	R6 = 0x0	R7 = 0x0
Register file thread id 6:
	R0 = 0x0	R1 = 0x0	R2 = 0x0	R3 = 0x0	R4 = 0x0	R5 = 0x0	R6 = 0x0	R7 = 0x0
Finegrained Multithreading CPI for this program 1.183333

